### PR TITLE
react-1098-create-component-for-select-users-page

### PR DESCRIPTION
**Pull request by chadiegilnerubia**
Ticket ID:
https://github.com/nerubia/kh360-react/issues/1028

### CODE REVIEW

File: __tests__/pages/admin/skill-map-administrations/[id]/select/select-employees.spec.ts [hunk 1]
```diff
@@ -187,8 +187,6 @@ test.describe("Admin - Select Employees", () => {
           .getByRole("checkbox")
       ).toBeVisible()
 
-      await expect(page.getByRole("heading", { name: "Select Employees" })).toBeVisible()
```
Comment (emmaacha) [inline __tests__/pages/admin/skill-map-administrations/[id]/select/select-employees.spec.ts:190]: We don't have to remove this line

**Review by emmaacha — CHANGES_REQUESTED** [review]
All is good, but can we change the name of the files for `select-user` folder to `select-user-filter`, `select-user-footer`, and so on? I think this gives more description for what the component is.

File: __tests__/pages/admin/skill-map-administrations/[id]/select/select-employees.spec.ts (hunk 1 above)
Comment (chadiegilnerubia) [reply __tests__/pages/admin/skill-map-administrations/[id]/select/select-employees.spec.ts]: Donee :3

**Comment by chadiegilnerubia** [comment]
> All is good, but can we change the name of the files for `select-user` folder to `select-user-filter`, `select-user-footer`, and so on? I think this gives more description for what the component is.

done :3

File: src/components/shared/select-user/select-user-filter.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import { useSearchParams } from "react-router-dom"
+import { useAppDispatch } from "@hooks/useAppDispatch"
+import { getUsers } from "@redux/slices/users-slice"
+
+interface FilterProps {
+  searchParamsKey: string
+  defaultStatus?: string
+}
+
+const SelectFilter = ({ searchParamsKey, defaultStatus = "all" }: FilterProps) => {
```
Comment (emmaacha) [inline src/components/shared/select-user/select-user-filter.tsx:15]: This should be `SelectUsersFilter`

File: src/components/shared/select-user/select-user-footer.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import { Button } from "@components/ui/button/button"
+import { Icon } from "@components/ui/icon/icon"
+import { CustomDialog } from "@components/ui/dialog/custom-dialog"
+import {
+  EvaluationAdministrationStatus,
+  type EvaluationAdministration,
+} from "@custom-types/evaluation-administration-type"
+import { type SurveyAdminstration } from "@custom-types/survey-administration-type"
+
+interface SelectFooterProps {
```
Comment (emmaacha) [inline src/components/shared/select-user/select-user-footer.tsx:10]: This should be `SelectUsersFooter`

File: src/components/shared/select-user/select-user-header.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { PageTitle } from "@components/shared/page-title"
+
+interface SelectHeaderProps {
+  title: string
+}
+
+export const SelectHeader = ({ title }: SelectHeaderProps) => {
```
Comment (emmaacha) [inline src/components/shared/select-user/select-user-header.tsx:7]: This should be `SelectUsersHeader`

File: src/components/shared/select-user/select-user-table.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
+import React from "react"
+import { Checkbox } from "@components/ui/checkbox/checkbox"
+import { Pagination } from "@components/shared/pagination/pagination"
+import { formatDate } from "@utils/format-date"
+import { type User } from "@custom-types/user-type"
+import { type EvaluationResult } from "@custom-types/evaluation-result-type"
+import { type SurveyResult } from "@custom-types/survey-result-type"
+
+interface Props {
```
Comment (emmaacha) [inline src/components/shared/select-user/select-user-table.tsx:9]: I think this should be `SelectUsersTableProps`

File: src/components/shared/select-user/select-user-table.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  selectedEmployeeIds: number[]
+  selectResults: Array<EvaluationResult | SurveyResult> // Union type
+  handleSelectAll: (checked: boolean) => void
+  handleClickCheckbox: (checked: boolean, employeeId: number) => void
+  hasPreviousPage: boolean
+  hasNextPage: boolean
+  totalPages: number
+}
+
+const SelectTable: React.FC<Props> = ({
```
Comment (emmaacha) [inline src/components/shared/select-user/select-user-table.tsx:20]: This should be `SelectUsersTable`

**Review by emmaacha — CHANGES_REQUESTED** [review]
Sorry for being nitpicky with the names 😅 File names should also be select "users" filter, etc. pala, my bad!

**Review by emmaacha — APPROVED** [review]

**Pull request merged**